src/lib.rs

```diff
@@ -14,14 +14,16 @@
 //!
 //! ## Example
 //!
+//! *Demonstration alongside the [`Error`][std::error::Error] derive macro from [`thiserror`](https://docs.rs/thiserror/1.0.25/thiserror/index.html),
+//! to propagate source locations from [`io::Error`][std::io::Error] with the `#[source]` attribute:*
 //! ```rust
 //! use std::io;
 //! use displaydoc::Display;
 //! use thiserror::Error;
 //!
 //! #[derive(Display, Error, Debug)]
 //! pub enum DataStoreError {
-//!     /// data store disconnected
+//!     /// data store disconnected: {0:?}
```

Comment:
This would be an incorrect usage. https://github.com/yaahc/blog.rust-lang.org/blob/master/posts/inside-rust/2021-05-15-What-the-error-handling-project-group-is-working-towards.md#duplicate-information-issue

We recommend either returning an error as a source or including it in your error message, but never both.

```suggestion
//!     /// data store disconnected
```

Reply:
Oh this is awesome!! Thanks so much for this work on standardizing errors!!! Really liked the part about the priorities of reporting vs downcasting.